schema/arrow.go

```diff
@@ -0,0 +1,393 @@
+package schema
+
+import (
+	"fmt"
+
+	"github.com/goccy/go-json"
+
+	"github.com/apache/arrow/go/v12/arrow"
+	"github.com/apache/arrow/go/v12/arrow/array"
+	"github.com/apache/arrow/go/v12/arrow/memory"
+	"github.com/cloudquery/plugin-sdk/types"
+)
+
+const (
+	MetadataUnique         = "cq:extension:unique"
+	MetadataPrimaryKey     = "cq:extension:primary_key"
+	MetadataIncremental    = "cq:extension:incremental"
+
+	MetadataTrue           = "true"
+	MetadataFalse          = "false"
+	MetadataTableName      = "cq:table_name"
+)
+
+type FieldChange struct {
+	Type       TableColumnChangeType
+	ColumnName string
+	Current    arrow.Field
+	Previous   arrow.Field
+}
+
+type MetadataFieldOptions struct {
+	PrimaryKey bool
+	Unique     bool
+}
+
+type MetadataSchemaOptions struct {
+	TableName string
+}
+
+func NewSchemaMetadataFromOptions(opts MetadataSchemaOptions) arrow.Metadata {
+	keys := make([]string, 0)
+	values := make([]string, 0)
+	if opts.TableName != "" {
+		keys = append(keys, MetadataTableName)
+		values = append(values, opts.TableName)
+	}
+	return arrow.NewMetadata(keys, values)
+}
+
+func NewFieldMetadataFromOptions(opts MetadataFieldOptions) arrow.Metadata {
+	keys := make([]string, 0)
+	values := make([]string, 0)
+	if opts.PrimaryKey {
+		keys = append(keys, MetadataPrimaryKey)
+		values = append(values, MetadataTrue)
+	}
+	if opts.Unique {
+		keys = append(keys, MetadataUnique)
+		values = append(values, MetadataTrue)
+	}
+
+	return arrow.NewMetadata(keys, values)
+}
+
+func MdIsPk(md arrow.Metadata) bool {
```

Comment:
```suggestion
func MetadataIsPk(md arrow.Metadata) bool {
```

Maybe easier to understand like this? Not sure if we can embed the struct and add the method like that